商品情報編集機能

DKannno opened the pull request:
#what
商品情報編集機能の実装
#why
商品情報編集ページに元々の情報を表示
商品情報の変更機能を追加

#（変更）商品情報の変更すると詳細ページへ遷移する　gyazo gif
https://gyazo.com/c24fd9913440807edba7f2402dabd515
#編集画面を開いた時点で元々の情報を表示　gyazo gif
https://gyazo.com/905a662f97f540ce030bf301e3dfbb81

DKannno committed:
商品編集ページの表示

DKannno committed:
データの編集機能実装（update)

DKannno committed:
変更するボタンをクリックするとトップページへ遷移

syuji-yamamoto commented on app/controllers/items_controller.rb:
```diff
@@ -21,7 +21,17 @@ def create
-  
+
+  def edit
+    @item = Item.find(params[:id])
```
```@item = Item.find(params[:id])```
同じ記述を複数箇所で使用しているので、一つにまとめてbefore_actionで使用しましょう！

DKannno replied:
修正しました。

syuji-yamamoto commented on app/controllers/items_controller.rb:
```diff
@@ -21,7 +21,17 @@ def create
+  end
+
+  def update
+    item = Item.find(params[:id])
```
```item = Item.find(params[:id])```
こちらローカル変数をインスタンス変数に変更すれば、before_actionでひとまとめにしてシンプルに記述することができますので修正しましょう。

DKannno replied:
修正しました。

syuji-yamamoto commented on app/controllers/items_controller.rb:
```diff
@@ -21,7 +21,17 @@ def create
+
+  def update
+    item = Item.find(params[:id])
+    item.update(item_params)
```
こちら、失敗した時も、次の行が実行されてしまいます。
if文で条件分岐して、前の画面を表示するなどのエラーハンドリングをしておきましょう！
※エラーハンドリングとは、プログラムの処理中正常に動かなかった場合に備え、その処理をエラーとして対処する処理のことです。

DKannno replied:
修正しました。

syuji-yamamoto commented on app/views/items/edit.html.erb:
```diff
@@ -7,10 +7,10 @@ app/assets/stylesheets/items/new.css %>
     <%# インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
-    <%= render 'shared/error_messages', model: f.object%>
+    <%= render 'shared/error_messages', model: f.object %>
     <%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
```
実装が完了したら、可読性向上のためコメントアウトは削除しておきましょう！

DKannno replied:
修正しました。

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認ください！

DKannno committed:
befoer_actionでインスタンスをまとめる

@item = Item.find(params[:id])をbefore_actionでまとめる。show,edit,updateのアクションの前に時効されるset_itemメソッドを定義。

DKannno committed:
更新時のエラーハンドリング設定。

正常に更新されない場合はトップ画面には遷移せず、再び編集画面へ。

DKannno committed:
不要なコメントアウトを削除

DKannno committed:
商品情報更新したら商品詳細ページへ遷移に変更

更新したらトップページへの遷移でしたが、商品詳細ページへの遷移へ変更。

yupimaru reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！
